GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -100,7 +100,7 @@ public void UpdateSubmodulesStatus(bool updateStatus, string workingDirectory, s
                 // First task: Gather list of submodules on a background thread.
 
                 // Don't access Module directly because it's not thread-safe.  Use a thread-local version:
-                var threadModule = new GitModule(workingDirectory);
+                var threadModule = gitModule;
```

Comment:
I don't like this change. I don't like passing an instance of `GitModule` around because it is not thread safe.
Line 102 also states "Don't access Module directly because it's not thread-safe.  Use a thread-local version".

Reply:
This is how the thread local version is created everywhere else, like in GitStatusMonitor. Before the fix up commit an optimized copy was used instead, we can apply that everywhere modules are used in threads instead.

Reply:
Moved this to #6373